MailerQ/RestApi/Client/IRestApiClient.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MailerQ.RestApi.Client
+{
+    /// <summary>
+    /// Contract that represent the HTTP methods availables in the MailerQ Rest API
+    /// </summary>
+    public interface IRestApiClient
+    {
+        /// <summary>
+        /// Request an HTTP Get to obtain a collection of instances of the model
+        /// </summary>
+        /// <typeparam name="T">Model type that return the endpoint</typeparam>
+        /// <param name="cancellationToken">The request cancellationToken</param>
+        /// <returns>Collection of objects returned by the endpoint. Null if http status code of the request is different than OK</returns>
+        Task<IRestApiClientResponse<T>> Get<T>(CancellationToken cancellationToken = default) where T : IRestApiModel;
+
+        /// <summary>
+        /// Request an HTTP Delete remove an instance of the model
+        /// </summary>
+        /// <typeparam name="T">Model type that accept the endpoint</typeparam>
+        /// <param name="model">Object model to send in the request body</param>
+        /// <param name="cancellationToken">The request cancellationToken</param>
+        /// <returns>The HTTP status code returned on the request</returns>
+        Task<IRestApiClientResponse> Delete<T>(T model, CancellationToken cancellationToken = default) where T : IRestApiModel;
+
+        /// <summary>
+        /// Request an HTTP Post create or update an instance of the model
+        /// </summary>
+        /// <typeparam name="T">Model type that accept the endpoint</typeparam>
+        /// <param name="model">Object model to send in the request body</param>
+        /// <param name="cancellationToken">The request cancellationToken</param>
+        /// <returns>The HTTP status code returned on the request</returns>
```

Comment:
@ms-darianbenito please change the return description

Reply:
Done! I changed all return descriptions